### ISSUE TITLE
Model-family coverage: 35 models declare a category without implementing its interface

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet#2138]
Split out of #2091, which established that the model-family coverage gap is **338 models, not the ~1170 originally reported** (real coverage is 80.7%, not ~36% — see https://github.com/ooples/AiDotNet/issues/2091#issuecomment-5604509217).

Every uncovered model now carries an unambiguous, fully-qualified `ADNGEN001` naming its specific reason. This issue covers one of the three resulting buckets.

## The bucket

Models that **declare a `[ModelCategory]` but do not implement that family's interface**, so the generated fixture cannot build them:

| Family | Required interface | Models |
|---|---|---:|
| `NeuralNetwork` | `INeuralNetworkModel<T>` | 10 |
| `Diffusion` | `IDiffusionModel<T>` | 17 |
| `TimeSeries` | `UsesMatrixInput && UsesVectorOutput` | 6 |
| `GAN` | `INeuralNetworkModel<T>` | 2 |

## Two distinct causes, needing different fixes

**1. Descriptive category hijacking routing.** `[ModelCategory(Diffusion)]` is frequently documentation of *how the model works* rather than a structural contract. `MatchaTTS`, `CosyVoice2` and `StyleTTS2` derive from `AudioNeuralNetworkBase<T>`, and `EmuEdit`, `MGIE` and `SmartEdit` from `VisionLanguageModelBase<T>` — both of which derive from `NeuralNetworkBase<T> : INeuralNetworkModel<T>`. These models **do** satisfy a neural-network family; they were routed to `Diffusion` purely by the descriptive category.

This is the same defect already fixed at `ResolveTestBaseClass` priority 18, where `UsesTensorInput` was treated as evidence of being a neural network. The category stays as accurate documentation; only the family choice changes.

**2. Genuine model defects.** Where the category *is* structural and the interface really is absent:

- `YOLOv9`, `FasterRCNN` — `[ModelCategory(NeuralNetwork)]`, derive from `ObjectDetectorBase<T> : ModelBase<...>`, no `INeuralNetworkModel`
- `EGarchModel`, `Garch11Model`, `GjrGarchModel` — `[ModelCategory(TimeSeriesModel)]`, derive from `ClassicalVolatilityModelBase<T>`, wrong I/O shape for the family fixture
- `AnoGANDetector`, `GANomalyDetector` — `[ModelCategory(GAN)]`, derive from `AnomalyDetectorBase<T>`

For these the fix is per model: implement the interface, or correct the category where the model genuinely is not what it claims.

## Approach

Central routing fix first, then case-by-case on whatever genuinely remains. Category metadata is preserved throughout — it is accurate, and stripping it would lose real information about how these models work.

## Acceptance

- [ ] Models that satisfy a family are routed to it rather than to one they cannot satisfy
- [ ] `TestedCount` rises from the 1466 baseline, gated by `ModelCoverageBaselineTests`
- [ ] Each remaining model has either the interface implemented or its category corrected, with the reasoning recorded
- [ ] No model's `[ModelCategory]` is removed merely to silence a diagnostic

Refs #2091

**Comment by ooples** [comment]
## Progress: two real model defects fixed, plus a routing correction that needs a second look

Branch `fix/2091-coverage-instrument`, draft PR #2136.

### The routing fix gained 16 models

`ResolveTestBaseClass` priority 4 routed on `[ModelCategory(Diffusion)]` alone into a fixture requiring `IDiffusionModel`. For models where that category is *descriptive* — they use diffusion internally but derive from `NeuralNetworkBase<T> : INeuralNetworkModel<T>` — it handed them a family they cannot satisfy. Coverage **1466 → 1482 (80.7% → 81.6%)**, sixteen newly covered, none lost, every other `ADNGEN001` bucket unchanged.

### Two genuine model defects, found the first time an invariant ran

**MatchaTTS — inert model.** `CreateDefaultMatchaTTSLayers` emitted the duration predictor *in series*: `FullyConnectedLayer(textEncoderDim)` then `FullyConnectedLayer(1)`, immediately before the flow-matching decoder. `PredictCore` is a plain sequential fold, so every input collapsed through **one scalar** and the model produced all-zero output for any input. In Matcha-TTS the duration predictor is a parallel head that never feeds the decoder. **1 failing → 33 passing.**

The diagnostic worth reusing: an L2 distance of *exactly* `0.000E+000` means information is destroyed structurally, not that training collapsed — a genuinely collapsed model gives a tiny non-zero distance. I first blamed an early `LayerNormalization` dividing out the scale of the constant-fill probe inputs, which would have made the **fixture** at fault. A probe on an **untrained** model refuted it: outputs were already identical *and all zero* before a single training step.

**EmuEdit / MGIE / SmartEdit — missing input projection.** `CreateDefaultEditingInstructionLayers` opened with `LayerNorm` straight into attention at `visionDim`, so it demanded pre-embedded tokens while all three declare `ImageSize` and are built with an image architecture. Every other stack in the file opens with a projection. **MGIE: 28 failing → 5.** The remaining five are a separate defect; EmuEdit and SmartEdit are not yet re-verified.

### The five finance diffusion models are routed wrong — by my own fix

`CSDI` (9 failures), `ScoreGrad` (5), `DiffusionTS` (4). CSDI's `Predict_ShouldBeDeterministic` reports −82828 vs −152044 across repeated inference.

That is **correct behaviour**: CSDI is a conditional score-based diffusion model with `SampleStandardNormal(Random)` and a noise schedule. My routing fix moved these out of the Diffusion family (where they had no fixture) into a neural-network family whose invariants *demand determinism*.

The fix is right for the TTS and editing models — deterministic networks that merely use diffusion internally — and wrong for these. **The correct answer is for them to implement `IDiffusionModel`** (`Scheduler`, `Generate`, `PredictNoise`, `ComputeLoss`) so priority 4 routes them to `DiffusionModelTestBase`, which understands stochastic sampling. That is per-model work on ~7 models.

### A negative result worth recording

I scanned all 451 default layer stacks for the two defect shapes. The first pass flagged 94 — almost all false positives, because `yield return layer` matched a *variable name* and conv/graph/recurrent first layers consume input legitimately. Narrowed to the real signature it gave 10 candidates; **all 10 verified clean**. `Pengi`, `OpenCLIP` and `BiaffineNER` pass 102/106, and `UnifiedMultimodal`'s narrow heads are parsed by index rather than folded. So beyond the editing trio and MatchaTTS, no further instances of either defect exist in the library.

### Remaining on this issue

- 7 finance/probabilistic diffusion models to implement `IDiffusionModel`
- MGIE's residual 5 failures; EmuEdit and SmartEdit re-verification
- A patch embedding instead of the flat `Dense(49152 → 1024)` projection, which makes these fixtures very slow
- The 10 `NeuralNetwork`, 6 `TimeSeries` and 2 `GAN` mismatches not yet touched

**Comment by ooples** [comment]
## The editing trio's remaining failures are a fixture/model contract disagreement, not a layer bug

Three successive layer-stack changes each fixed one thing and revealed another, which usually means the *contract* is undefined rather than the implementation being wrong in three places. It is.

`SmartEdit` documents its output contract in its own summary:

> Output: edited image tensor of size `OutputImageSize * OutputImageSize * 3`

The generated fixture declares:

```csharp
protected override int[] InputShape => new[] { 3, 128, 128 };
protected override int[] OutputShape => new[] { 4 };
=> new SmartEdit<float>(new NeuralNetworkArchitecture<float>(..., outputSize: 4));
```

An image editor is being asked to produce **4 values**. The same holds for `EmuEdit` and `MGIE`. Their output width comes from `_options.OutputImageSize`, and their embedding width from `_options.VisionDim` — neither derives from the architecture the fixture constructs, so architecture and options disagree by construction.

### What this means for the three changes made so far

- **Input projection — correct, keep.** The stack genuinely could not accept its own declared input; every other default stack in `LayerHelper` opens with a projection and this one did not. MGIE went 28 failing → 5.
- **Encoder boundary — correct, keep.** A necessary consequence: the models index off the stack's layout and the leading-layer count changed. This was my regression from the previous commit, corrected.
- **Task head — wrong, reverted.** I appended `Dense(Architecture.OutputSize)` copying the `Oscar`/`UNITER` convention. That convention is for a *classification* head; these are image editors. SmartEdit went from 3 failures to 7, including new ones (`Clone_ShouldProduceIdenticalOutput`, `NamedLayerActivations_ShouldBeNonEmpty`). Reverted rather than committed.

### The actual fix

The generator must construct these models with an architecture that agrees with their options, rather than guessing `outputSize: 4` and `128x128x3`. The mechanism already exists and is used: the `SEEDX` case constructs with explicit scaled options and a matching architecture, commented as *"ImageSize and the architecture stay at 128 so the emitted InputShape still matches the forward path"*.

That is a generator-side fix, not a per-model edit, and it is the same class of problem as the `InputShape` override defeating `NeuralNetworkModelTestBase`'s model-derived default.

I am not making further layer changes to these three until the geometry is settled, because each speculative change so far has moved the failure rather than removed it.
